client_test.go

```diff
@@ -1071,15 +1071,21 @@ func Test_Client_StopAndCancel(t *testing.T) {
 
 		client, _ := setup(t)
 
+		// A reference to the stopped channel should be taken before calling Stop.
+		stopped := client.Stopped()
```

Comment:
Making `Client.Stopped` a pass through to `BaseStartStop.Stopped` does now require that a reference to the `Stopped` channel be taken _before_ calling `Stop` or `StopAndCancel`. Calling it after bears the risk that you might be getting the _next_ `Stopped` channel which is being preemptively initialized for the next time the client is `Started`.

This does have the advantage though that you can take a reference to `Stopped` before calling a client's `Start`, something that previously didn't work.

Reply:
(NM, changed this.)